merino/providers/adm.py

```diff
@@ -160,7 +162,9 @@ async def _fetch(self) -> None:
         """Fetch suggestions, keywords, and icons from Remote Settings."""
         # A dictionary keyed on suggestion keywords, each value stores an index
         # (pointer) to one entry of the suggestion result list.
-        suggestions: dict[str, int] = {}
+        suggestions: dict[str, Tuple[int, int]] = {}
+        # A list of full keywords
+        full_keywords_list: list[str] = []
```

Comment:
Nit: to be consistent with other variables, let's drop the suffix `list`.
```suggestion
        full_keywords: list[str] = []
```